app/team_test.go

```diff
@@ -96,13 +96,33 @@ func TestAddUserToTeam(t *testing.T) {
 	th := Setup().InitBasic()
 	defer th.TearDown()
 
-	user := model.User{Email: strings.ToLower(model.NewId()) + "success+test@example.com", Nickname: "Darth Vader", Username: "vader" + model.NewId(), Password: "passwd1", AuthService: ""}
-	ruser, _ := th.App.CreateUser(&user)
+	t.Run("add user", func(t *testing.T) {
```

Comment:
Can you add some coverage for
* `th.BasicTeam.AllowedDomains` is non-empty but the user domain matches
* `th.BasicTeam.AllowedDomains` has multiple domains
* `th.BasicTeam.AllowedDomains` is non-empty and the user has a subdomain of a valid domain? (Should reject it.)
* Tests that include setting `RestrictCreationToDomains` and the interplay between both (e.g. team override, but system disallows).

Also, it looks like there's some coverage in `api4/team_test.go` that tests `RestrictCreationToDomains`. It's unfortunate the way our code is split up, but I suspect it makes the most sense to co-locate these tests together. 0/5 on whether we extend the tests here or add to the tests in `api4/team_test.go`.